server/server.go

```diff
@@ -662,3 +634,43 @@ func (s *Server) resolveAuthorizationModelID(ctx context.Context, store, modelID
 
 	return modelID, nil
 }
+
+// HealthChecker defines an interface that services can implement for server health checks.
+type HealthChecker interface {
+	IsReady(ctx context.Context) (bool, error)
+}
+
+type openfgaHealthServer struct {
+	healthv1pb.UnimplementedHealthServer
+	HealthChecker
+}
+
+var _ grpc_auth.ServiceAuthFuncOverride = (*openfgaHealthServer)(nil)
+
+// AuthFuncOverride implements the grpc_auth.ServiceAuthFuncOverride interface by bypassing authn middleware.
+func (o *openfgaHealthServer) AuthFuncOverride(ctx context.Context, fullMethodName string) (context.Context, error) {
```

Comment:
Attached this directly to the `openfgaHealthServer` instead of the intermediate type composition.